### PR TITLE
Scheduler manager/Issue-87

### PR DESCRIPTION
**Pull request by TGrigor**
Added generic repository with entity structures.

### CODE REVIEW

File: src/SchedulerManager/Travely.SchedulerManager.Repository/SchedulerDbContext.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        public virtual DbSet<MessageTemplate> MessageTemplates { get; set; }
+        public virtual DbSet<ScheduleInfo> ScheduleInfos { get; set; }
+        public virtual DbSet<UserSchedule> UserSchedules { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<ScheduleInfo>(entity =>
+            {
```
Comment (ashpetrosyan97) [inline src/SchedulerManager/Travely.SchedulerManager.Repository/SchedulerDbContext.cs:24]: Enitityneri valdationneri attributnery karelia hanel sax grel contextum, datark constructor mez voncor te petq chi

File: src/SchedulerManager/Travely.SchedulerManager.Repository/Implementation/Repository.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            _context = context;
+            _dbSet = context.Set<TEntity>();
+        }
+        public ValueTask<TEntity> FirstAsync(int id)
+        {
+            return _dbSet.FindAsync(id);
+        }
+        public async Task<IEnumerable<TEntity>> GetAsync(Expression<Func<TEntity, bool>> predicate)
+        {
+            return await _dbSet.Where(predicate).ToListAsync();
```
Comment (ashpetrosyan97) [inline src/SchedulerManager/Travely.SchedulerManager.Repository/Implementation/Repository.cs:27]: GetAsyncy aranc AsNoTracking es vercrel vtangavor a, petq a haskanal mez misht a petq track anenq mer entitynery te che

**Review by ashpetrosyan97 — APPROVED** [review]

**Review by Tavrizyan — APPROVED** [review]

File: src/SchedulerManager/Travely.SchedulerManager.Repository/Implementation/Repository.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        }
+        public void Update(TEntity entity)
+        {
+            _dbSet.Update(entity);
+        }
+        public Task<bool> AnyAsync(Expression<Func<TEntity, bool>> predicate)
+        {
+            return _dbSet.AnyAsync(predicate);
+        }
+        public async Task AddAsync(TEntity entity)
```
Comment (artonoyan) [inline src/SchedulerManager/Travely.SchedulerManager.Repository/Implementation/Repository.cs:37]: I'm not sure that you need `async-await`, you can return a `Task`

File: src/SchedulerManager/Travely.SchedulerManager.Repository/Entities/ScheduleInfo.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            UserSchedules = new HashSet<UserSchedule>();
+        }
+
+        [Key]
+        public int Id { get; set; }
+        public int MessageTemplateId { get; set; }
+        public DateTime ExpirationDate { get; set; }
+        public int RecurseId { get; set; }
+        [Required]
```
Comment (artonoyan) [inline src/SchedulerManager/Travely.SchedulerManager.Repository/Entities/ScheduleInfo.cs:20]: Why you do not use [fluent API](https://docs.microsoft.com/en-us/ef/core/modeling/)?

**Review by anush1111 — APPROVED** [review]

**Review by ashpetrosyan97 — COMMENTED** [review]
new DateTime() -> DateTime.NowUtc
SingleAsync  -> FirstOrDefaultAsync()
When inserting we do not need initialize ModifiedOn 
Active column is for soft delete  more relevant to  use IsDeleted name for it
.Where(s => s.Active) more relevant to use QueryFilter
.AsQueryable()  is redundant, DbSet<T> implements IQueryable<T>

**Comment by TGrigor** [comment]
> new DateTime() -> DateTime.NowUtc
> SingleAsync -> FirstOrDefaultAsync()
> When inserting we do not need initialize ModifiedOn
> Active column is for soft delete more relevant to use IsDeleted name for it
> .Where(s => s.Active) more relevant to use QueryFilter
> .AsQueryable() is redundant, DbSet implements IQueryable

**Review by TatevAsatryan — DISMISSED** [review]

**Review by artonoyan — APPROVED** [review]

**Review by anush1111 — DISMISSED** [review]

**Review by anush1111 — APPROVED** [review]

**Review by artonoyan — APPROVED** [review]

**Pull request merged**